k4MarlinWrapper/k4MarlinWrapper/util/k4MarlinWrapperUtil.h

```diff
@@ -20,6 +23,13 @@ namespace k4MW::util {
     std::regex re{"\\s+"};
     return split(subject, re);
   }
+
+  /// singleton helper to intialise global Marlin parameters exactly once
+  const static marlin::StringParameters* marlin_global_parameters = ([]() {
+    marlin::Global::parameters = new marlin::StringParameters;
+    return marlin::Global::parameters;
+  })();
```

Comment:
This is unfortunately apparently not enough to guarantee that the global parameters are initialized only (and exactly) once. This is also causing the test to fail, as this initialization now happens twice, therefore:
- the global parameters no longer have a `RandomSeed` stored
- `StringParameters::getIntVal` returns `0` if a key is not found

In this way the random seed is reset to a different value, causing the test to fail.

Reply:
OK. The reason this is not working as expected is because a global static variable signifies a variable that is only visible inside one translation unit. Given that this header is included in two translation units, we have a double initialization. Function scope statics on the other hand will always refer to the same variable, which is why the initialization as implemented now works.

Reply:
You fixed this issue, and things are OK now as is, right?

Reply:
Yes, that is correct. The `marlin::Global::parameters` are now correctly initialized (thread safe even), and as long as the `k4MarlinWraperUtil.h` header is included in a translation unit, they can be used directly anywhere (that on the other hand is only thread safe for reading)